packages/adapter-drizzle-all/src/mysql/index.ts

```diff
@@ -0,0 +1,293 @@
+        .select()
+        .from(users)
+        .where(eq(users.id, data))
+        .then(res => res[0])
+        ?? null
+    },
+    getUserByEmail: async (data) => {
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.email, data))
+        .then(res => res[0])
+        ?? null
+    },
+    createSession: async (data) => {
+      await client
+        .insert(sessions)
+        .values(data)
+
+      return client
+        .select()
+        .from(sessions)
+        .where(
+          eq(
+            sessions.sessionToken,
+            data.sessionToken
+          )
+        )
+        .then(res => res[0])
+    },
+    getSessionAndUser: async (data) => {
+      return client.select({
+        session: sessions,
+        user: users
+      })
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data))
+        .innerJoin(users, eq(users.id, sessions.userId))
+        .then(res => res[0])
+        ?? null
+    },
+    updateUser: async (data) => {
+      if (!data.id) {
+        throw new Error("No user id.")
+      }
+
+      await client
+        .update(users)
+        .set(data)
+        .where(eq(users.id, data.id))
+
+      return client
+        .select()
+        .from(users)
+        .where(
+          eq(
+            users.id,
+            data.id
+          )
+        )
+        .then(res => res[0])
+
+
+    },
+    updateSession: async (data) => {
+      await client
+        .update(sessions)
+        .set(data)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+
+      return client
+        .select()
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+        .then(res => res[0])
+    },
+    linkAccount: async (rawAccount) => {
+      await client
+        .insert(accounts)
+        .values(rawAccount)
+        .then(res => res[0])
+    },
+    getUserByAccount: async (account) => {
+      return client.select({
+        id: users.id,
+        email: users.email,
+        emailVerified: users.emailVerified,
+        image: users.image,
+        name: users.name
+      })
+        .from(users)
+        .innerJoin(accounts, (
+          and(
+            eq(accounts.providerAccountId, account.providerAccountId),
+            eq(accounts.provider, account.provider)
+          )
+        ))
+        .then(res => res[0])
+        ?? null
+    }
+    ,
+    deleteSession: async (sessionToken) => {
+      await client
+        .delete(sessions)
+        .where(eq(sessions.sessionToken, sessionToken))
+
+    },
+    createVerificationToken: async (token) => {
+      await client
+        .insert(verificationTokens)
+        .values(token)
+
+      return client
+        .select()
+        .from(verificationTokens)
+        .where(eq(verificationTokens.identifier, token.identifier))
+        .then(res => res[0])
+    },
+    useVerificationToken: async (token) => {
+      try {
+        await client
+          .delete(verificationTokens)
```

Comment:
Shouldn't we select the token for later return first before deleting it?

Reply:
Oops, yes!